doc/source/whatsnew/v0.25.0.rst

```diff
@@ -167,7 +167,7 @@ Numeric
 Conversion
 ^^^^^^^^^^
 
--
+- Bug in :meth:`localize_pydatetime` where some ``datetime`` were incorrectly treated as pandas :class:`Timestamp` (:issue:`25734`)
```

Comment:
This should describe the main user facing impact that was in the original issue; arithmetic with `CustomBusinessDay` and datetime subclasses.